packages/manager/modules/nasha/src/partition/access/controller.js

```diff
@@ -8,165 +8,146 @@ export default class NashaPartitionAccessCtrl {
+        }).$promise,
+      accesses: this.OvhApiDedicatedNasha.Partition().Access().v6()
+        .query({
+          serviceName: this.serviceName,
+          partitionName: this.partition,
+        }).$promise,
+    }).then((data) => {
+      this.data.nasha = data.nasha;
+      this.data.partition = data.partition;
+      this.table.accessIps = data.accesses.map(ip => ({
+        ip,
+      }));
+      if (resetCache) {
+        this.table.refresh = !this.table.refresh;
+      }
+      if (this.isNew) {
+        this.table.accessIps.push(this.access);
+        this.data.addAccessInProgress.push(this.access);
+      }
+    }).catch((err) => {
+      this.CucCloudMessage.error(this.$translate.instant('nasha_partitions_access_no_data_error'));
+      return this.$q.reject(err);
+    }).finally(() => {
+      this.loaders.table = false;
+    });
+  }
 
-    self.addAccess = function addAccess() {
-      self.openModal('nasha/partition/access/add/template.html', 'NashaPartitionAccessAddCtrl', {
-        serviceName: self.data.nasha.serviceName,
-        partition: self.data.partition,
-      });
-    };
 
-    /*= =====================================
-       =                Polling              =
-       ====================================== */
-
-    function launchPolling(taskId) {
-      return Poller.poll(`/dedicated/nasha/${self.data.nasha.serviceName}/task/${taskId}`,
-        null,
-        {
-          successRule(task) {
-            return task.status === 'done';
-          },
-          errorRule(task) {
-            return ['doing', 'todo', 'done'].indexOf(task.status) === -1;
-          },
-          namespace: 'nasha.access',
-        });
+  getAccessForIp(accessIp) {
+    // If the access is being added, return the local data
+    const accessAddInProgress = find(this.data.addAccessInProgress, item => item.ip === accessIp);
+    if (accessAddInProgress) {
+      return accessAddInProgress;
     }
 
-    function pollTasksForAccess(access, taskId) {
-      launchPolling(taskId)
-        .finally(() => {
-          // Remove from the polling list
-          remove(self.data.taskForAccess, item => item.task === taskId);
-
-          // If the partition was in creation, remove it from the creation list
-          remove(self.data.addAccessInProgress, item => item.ip === access.ip);
+    // if not we get the details form the api
+    return this.OvhApiDedicatedNasha.Partition().Access().v6().get({
+      serviceName: this.serviceName,
+      partitionName: this.data.partition.partitionName,
+      ip: accessIp,
+    }).$promise.then(data => data);
+  }
 
-          self.updateAccess(access);
-        });
-    }
+  transformItem(access) {
+    return this.getAccessForIp(access.ip);
+  }
 
-    $scope.$on('$destroy', () => {
-      Poller.kill({ namespace: 'nasha.access' });
+  removeAccess(access) {
+    this.openModal('nasha/partition/access/delete/template.html', 'NashaPartitionAccessDeleteCtrl', {
```

Comment:
The purpose of the PR is to kill `uibModal` in NASHA, but it could also make sense to migrate this one as part of this scope